src/javascripts/events/domEvents.js

```diff
@@ -35,195 +37,234 @@ import showFoodReadOnly from '../components/readOnlyPrinters/showFoodReadOnly';
+      const firebaseKey = e.target.id.split('--')[1];
+      getEvents().then((x) => {
+        deleteShow(firebaseKey, x.firebaseKey).then((showsArray) => printShows(showsArray));
+      });
     }
-
-    if (e.target.id.includes('edit-staff')) {
+  }
+  if (e.target.id.includes('delete-staff')) {
+  // eslint-disable-next-line no-alert
+    if (window.confirm('Are you sure?')) {
       const firebaseKey = e.target.id.split('--')[1];
-      editFormModal('Update Staff');
-      getSingleStaff(firebaseKey).then((staffObject) => updateStaffForm(staffObject));
+      getEvents().then((x) => {
+        deleteStaff(firebaseKey, x.firebaseKey).then((staffArray) => showStaff(staffArray));
+      });
     }
+  }
 
-    if (e.target.id.includes('update-staff')) {
-      const firebaseKey = e.target.id.split('--')[1];
-      e.preventDefault();
+  if (e.target.id.includes('edit-staff')) {
+    const firebaseKey = e.target.id.split('--')[1];
+    editFormModal('Update Staff');
+    getSingleStaff(firebaseKey).then((staffObject) => updateStaffForm(staffObject));
+  }
+
+  if (e.target.id.includes('update-staff')) {
+    const firebaseKey = e.target.id.split('--')[1];
+    e.preventDefault();
+    getEvents().then((x) => {
       const staffObject = {
         first_name: document.querySelector('#staffFirstName').value,
         last_name: document.querySelector('#staffLastName').value,
         staff_image: document.querySelector('#staffImage').value,
         role: document.querySelector('#staffRole').value,
       };
-      updateStaff(firebaseKey, staffObject, id).then((staffArray) => showStaff(staffArray));
+      updateStaff(firebaseKey, staffObject, x.firebaseKey).then((staffArray) => showStaff(staffArray));
       $('#formModal').modal('toggle');
-    }
-    if (e.target.id.includes('edit-show')) {
-      const firebaseKey = e.target.id.split('--')[1];
-      editFormModal('Edit Show');
-      getSingleShow(firebaseKey).then((showObject) => editShowForm(showObject));
-    }
-    if (e.target.id.includes('update-show')) {
-      const firebaseKey = e.target.id.split('--')[1];
-      e.preventDefault();
+    });
+  }
+  if (e.target.id.includes('edit-show')) {
+    const firebaseKey = e.target.id.split('--')[1];
+    editFormModal('Edit Show');
+    getSingleShow(firebaseKey).then((showObject) => editShowForm(showObject));
+  }
+  if (e.target.id.includes('update-show')) {
+    const firebaseKey = e.target.id.split('--')[1];
+    e.preventDefault();
+    getEvents().then((x) => {
       const showObject = {
         name: document.querySelector('#showName').value,
         image: document.querySelector('#image').value,
         date: document.querySelector('#date').value,
         description: document.querySelector('#description').value,
       };
-      updateShow(firebaseKey, showObject, id).then((showsArray) => printShows(showsArray));
+      updateShow(firebaseKey, showObject, x.preventDefault).then((showsArray) => printShows(showsArray));
```

Comment:
@jrobinson0529 is this the one causing the issue? x.firebaseKey should solve???

Reply:
Fixed!